File: Java/Binary Search Tree.java
import java.util.*;
class BinaryTree {
    BinaryTree left;
    int data;
    BinaryTree right;
    BinaryTree(int value) {
        left = null;
        data = value;
        right = null;
    }
    public BinaryTree insert(BinaryTree root, int value) {
        if (root == null) {
            return new BinaryTree(value);
        }
        if (value > root.data) {
	        root.right = insert(root.right, value);
        }
        else if (value < root.data) {
	        root.left = insert(root.left, value);
        }
        return root;
    }
    public Boolean search(BinaryTree root, int value) {
        if (root.data == value) {
            return true;
        }
        if (value > root.data) {
	        return search(root.right, value);
        }
        else if (value < root.data) {
	        return search(root.left, value);
        }
        return false;
    }
    public void inOrder(BinaryTree root) {
        if (root != null) {
            inOrder(root.left);
            System.out.print(" " + root.data);
            inOrder(root.right);
        }
    }
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
        BinaryTree obj;
        int n, value;
        System.out.print("Enter the Number of Elements : ");
        n = s.nextInt();
        System.out.print("Enter the Root Value : ");
        value = s.nextInt();
        obj = new BinaryTree(value);
        for (int i=0; i<n-1; i++) {
    	    System.out.print("Enter the Element : ");
    	    value = s.nextInt();
    	    obj.insert(obj, value);
	    }
        System.out.print("\n\nDisplaying BST Elements in Inorder\n");
        obj.inOrder(obj);
        System.out.print("\n\nEnter the Search Node : ");
        value = s.nextInt();
        if (obj.search(obj, value)) {
    	    System.out.print(value + "is Found");
	    }
	    else {
		    System.out.print(value + " is not Found");
	    }
    }
}

File: C++/Binary Search Tree.cpp
#include <bits/stdc++.h>
using namespace std;
class BST {
	public:
		BST *left;
		int data;
    	BST *right;
    	BST(int value) {
    		left = NULL;
    		data = value;
    		right = NULL;
		}
		BST* insert(BST* root, int value) {
			if (!root) {
		        return new BST(value);
		    }
		    if (value > root->data) {
		        root->right = insert(root->right, value);
		    }
		    else if (value < root->data) {
		        root->left = insert(root->left, value);
		    }
		    return root;
		}
		BST* search(BST* root, int value) {
			if (root == NULL || root->data == value) {
		        return root;
		    }
		    if (value > root->data) {
		        return search(root->right, value);
		    }
		    else if (value < root->data) {
		        return search(root->left, value);
		    }
		}
		void inOrder(BST* root) {
			if (!root) {
		        return;
		    }
		    inOrder(root->left);
		    cout << root->data << " ";
		    inOrder(root->right);
		}
};
int main() {
    BST *obj, *root = NULL;
    int n,value;
    cout << "Enter the Number of Elements : ";
    cin>>n;
    int arr[n];
    cout << "Enter the Root Value :";
    cin>>value;
    root = obj->insert(root, value);
    for (int i=0; i<n-1; i++) {
    	cout << "Enter the Element : ";
    	cin>>value;
    	obj->insert(root, value);
	}
    cout << "Displaying BST Elements in Inorder"<<endl;
    obj->inOrder(root);
    cout << "\nEnter the Search Node : ";
    cin>>value;
    if (obj->search(root, value)) {
    	cout << value << "is Found";
	}
	else {
		cout << value << " is not Found";
	}
    return 0;
}
